cub/test/catch2_test_device_reduce_large_offsets.cu

```diff
@@ -0,0 +1,247 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "insert_nested_NVTX_range_guard.h"
+// above header needs to be included first
+
+#include <cub/device/device_reduce.cuh>
+#include <cub/thread/thread_operators.cuh>
+
+#include <thrust/iterator/counting_iterator.h>
+
+#include <cuda/std/__algorithm_>
+
+#include <cstdint>
+
+#include "catch2_test_device_reduce.cuh"
+#include "catch2_test_helper.h"
+#include "catch2_test_launch_helper.h"
+
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Reduce, device_reduce);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Sum, device_sum);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Min, device_min);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::ArgMin, device_arg_min);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Max, device_max);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::ArgMax, device_arg_max);
```

Comment:
I am confused about the scope of this PR. The PR claims to target `ArgMin` and `ArgMax`. Also, only benchmarks for those two are added. Why are tests for all reductions added then?

If `Reduce`, `Sum`, `Min`, and `Max` already support large inputs, maybe split off those test as a separate PR.

Reply:
@elstehle Why do we add more tests and the algos this PR modifies?

Reply:
OK, sounds good!